src/cache/cache_las.c

```diff
@@ -1489,12 +1504,78 @@ __wt_find_lookaside_upd(
 		if (upd_type == WT_UPDATE_BIRTHMARK && ignore_birthmark)
 			break;
 
-		/* It is okay to locate an older update in lookaside,
-		 * but we proceed only if we find a matching record. */
-		WT_ASSERT(session, start_ts >= las_timestamp);
+		/*
+		 * It is okay to locate an older update in lookaside,
+		 * but we proceed only if we find a matching record.
+		 */
 		if (start_ts != las_timestamp)
 			break;
 
+		/*
+		 * Keep walking until we get a non-modify update.
+		 * Once we get to that point, squash the updates together.
+		 */
+		if (upd_type == WT_UPDATE_MODIFY) {
+			while (upd_type == WT_UPDATE_MODIFY) {
+				WT_ERR(__wt_update_alloc(session,
+				    &las_value, &upd, &incr, upd_type));
+				if (i >= WT_MODIFY_ARRAY_SIZE) {
+					if (i == WT_MODIFY_ARRAY_SIZE)
+						listp = NULL;
+					WT_ERR(__wt_realloc_def(session,
+					    &allocated_bytes, i + 1, &listp));
+					if (i == WT_MODIFY_ARRAY_SIZE)
+						memcpy(
+						    listp, list, sizeof(list));
+				}
+				listp[i++] = upd;
+				WT_ERR(cursor->next(cursor));
+#ifdef HAVE_DIAGNOSTIC
+				/*
+				 * We shouldn't be crossing over to another LAS
+				 * page id or breaking any visibility rules
+				 * while doing this.
+				 */
+				WT_ERR(cursor->get_key(cursor, &las_pageid,
+				    &las_id, &las_counter, &las_key));
+				WT_ASSERT(session,
+				    las_pageid == ref->page_las->las_pageid);
+				WT_ERR(__wt_compare(session,
+				    S2BT(session)->collator,
+				    &las_key, key, &cmp));
+				WT_ASSERT(session, cmp == 0);
+#endif
+				/*
+				 * Make sure we use the underscore variants of
+				 * these variables. We need to retain the
+				 * timestamps of the original modify we saw.
+				 */
+				WT_ERR(cursor->get_value(
+				    cursor, &_las_txnid, &_las_timestamp,
+				    &_durable_timestamp, &_prepare_state,
+				    &upd_type, &las_value));
+				WT_ASSERT(session, __wt_txn_visible(
+				    session, _las_txnid, _las_timestamp));
+			}
+			WT_ASSERT(session, upd_type == WT_UPDATE_STANDARD);
+
+			/*
+			 * This value is a pointer/size pair pointing at a spot
+			 * in the cursor's buffer. Our modify function needs to
+			 * know to allocate its own buffer when applying
+			 * modifications so explicitly unset these.
+			 */
+			las_value.mem = NULL;
+			las_value.memsize = 0;
+			while (i > 0) {
+				WT_ERR(__wt_modify_apply_item(
```

Comment:
instead of ```cursor->session``` you can pass ```session``` that is locally available to this function. They are the same session I imagine.

Reply:
I think that `session` is the one that's doing the read txn whereas the one I'm passing in is a LAS session.

Reply:
Ahh! ya, you are right this is las cursor and hence not the same user session. I think It is better to call with the user session, might be useful when logging to know what session called modify.

Reply:
24f2eea

Makes sense. Switched to the user session.